src/Umbraco.Web/Models/Mapping/ContentMapperProfile.cs

```diff
@@ -88,12 +91,45 @@ internal class ContentMapperProfile : Profile
         }
     }
 
+    internal class CultureUpdateDateResolver : IValueResolver<IContent, ContentItemBasic<ContentPropertyBasic>, DateTime>
+    {
+        public DateTime Resolve(IContent source, ContentItemBasic<ContentPropertyBasic> destination, DateTime destMember, ResolutionContext context)
+        {
+            var culture = context.GetCulture();
+
+            //a culture needs to be in the context for a variant content item
+            if (culture == null)
+                return source.UpdateDate;
+
+            var pubDate = source.GetPublishDate(culture);
```

Comment:
When using the variant specific methods like `GetPublishDate` and `GetCultureName`, these should only be used when the content item actually varies by culture else they will return null. So first you need to check `source.ContentType..VariesByCulture()` and if that is true, then you can use GetPublishDate, else you need to use the normal `source.PublishDate` for invariant content.